### PR TITLE
Added the force fields necessary for Calvados2

### PR DESCRIPTION
**Pull request by ywitzky**
Calvados2 is a strongly coarse grained model for intrinsically disordered proteins (IDP) which requires a yukawa potential, an harmonic potential and a modified Lennard Jones potential.

### CODE REVIEW

**Comment by jgreener64** [comment]
Looks great. I just merged a major change to master, would you mind rebasing/merging on top of that? It will involve changes to the mixing functions and zero shortcuts, hopefully you can use the changes to `LennardJones` as a guide.

**Comment by ywitzky** [comment]
I should have done the changes you mentioned. Let me know if i forgot something.

File: src/interactions/lennard_jones.jl [hunk 1]
```diff
@@ -280,3 +290,163 @@ function potential(::LennardJonesSoftCore, r2, invr2, (σ2, ϵ, σ6_fac))
     six_term = σ2^3 * inv(r2^3 + σ2^3 * σ6_fac)
     return 4ϵ * (six_term ^ 2 - six_term)
 end
+
+
+@doc raw"""
+    AshbaughHatch(; cutoff, α, λ, p, use_neighbors, lorentz_mixing, weight_special,
```
Comment (jgreener64) [inline src/interactions/lennard_jones.jl:296]: Update the arguments.

File: src/interactions/lennard_jones.jl [hunk 2]
```diff
@@ -280,3 +290,163 @@ function potential(::LennardJonesSoftCore, r2, invr2, (σ2, ϵ, σ6_fac))
...
+
+
+@kwdef struct AshbaughHatchAtom{C, M, S, E, L}
+    index::Int = 1
+    charge::C = 0.0
+    mass::M =1.0u"g/mol"
+    σ::S =0.0u"nm"
+    ϵ::E =0.0u"kJ * mol^-1"
+    λ::L=1.0
+    solute::Bool = false
```
Comment (jgreener64) [inline src/interactions/lennard_jones.jl:352]: Remove `solute`.

File: src/interactions/lennard_jones.jl [hunk 3]
```diff
@@ -280,3 +290,163 @@ function potential(::LennardJonesSoftCore, r2, invr2, (σ2, ϵ, σ6_fac))
...
+
+The AshbaughHatch ($V_{\text{AH}}$) is a modified Lennard-Jones ($V_{\text{LJ}}$) 6-12 interaction between two atoms.
+
+The potential energy is defined by
+```math
+V_{\text{LJ}}(r_{ij}) = 4\varepsilon_{ij} \left[\left(\frac{\sigma_{ij}}{r_{ij}}\right)^{12} - \left(\frac{\sigma_{ij}}{r_{ij}}\right)^{6}\right] \\ 
+```
+
+```math
+V_{\text{AH}}(r_{ij}) =
```
Comment (jgreener64) [inline src/interactions/lennard_jones.jl:307]: As it stands I think we don't apply the VLJ(rc) term by default? So either remove it in the docstring or add it to the code, in which case probably remove the `cutoff` option and don't use the cutoff path.

File: src/interactions/coulomb.jl [hunk 4]
```diff
@@ -340,3 +341,104 @@ end
         return pe
     end
 end
+
+
+
+
+
+@doc raw"""
+    Yukawa(; cutoff, use_neighbors, weight_special, coulomb_const, force_units, energy_units)
```
Comment (jgreener64) [inline src/interactions/coulomb.jl:350]: Update arguments.

File: src/interactions/coulomb.jl [hunk 5]
```diff
@@ -340,3 +341,104 @@ end
...
+```math
+F(r_{ij}) = \frac{q_i q_j}{4 \pi \varepsilon_0 r_{ij}^2} \exp(-\kappa r_{ij})\left(\kappa r_{ij} + 1\right) \vec{r}_{ij}
+```
+"""
+@kwdef struct Yukawa{C, W, T, D} 
+    cutoff::C= NoCutoff()
+    use_neighbors::Bool= false
+    weight_special::W =1
+    coulomb_const::T = coulomb_const
+    kappa::D = kappa=1.0*u"nm^-1"
```
Comment (jgreener64) [inline src/interactions/coulomb.jl:370]: Repetition.

File: src/interactions/coulomb.jl [hunk 6]
```diff
@@ -340,3 +341,104 @@ end
...
+    if special
+        return f * dr * inter.weight_special
+    else
+        return f * dr
+    end
+end
+
+function force_divr(::Yukawa, r2, invr2, (coulomb_const, qi, qj, kappa))
+    r = sqrt(r2)
+    return (coulomb_const * qi * qj) / r^3 *exp(-kappa*r) * (kappa*r+1.0)
```
Comment (jgreener64) [inline src/interactions/coulomb.jl:419]: `+1` rather than `+1.0` to avoid promoting `Float32` to `Float64`.

**Review by jgreener64 — COMMENTED** [review]
Thanks, just minor stuff, only thing to think about is the cutoff point.

**Comment by ywitzky** [comment]
I added the remarks in the code and added some tests for the options "special" and "shortcut".

**Pull request merged**